crates/lib/src/transaction/paid_transaction.rs

```
@@ -26,14 +27,24 @@ pub async fn sign_transaction_if_paid(
     let margin = margin.unwrap_or(0.0);
     let required_lamports = (min_transaction_fee as f64 * (1.0 + margin)) as u64;
 
+    let oracle_client = OracleClient::new(rpc_client);
+
+    let sol_mint = Pubkey::from_str_const(SOL_MINT);
+
+    let token_price_info = oracle_client
+        .get_token_price(&sol_mint)
+        .await
+        .map(|price| TokenPriceInfo { price })
+        .unwrap_or(TokenPriceInfo { price: 0.0 });
```

Instead of a 0 price we should return an error

---

Sure will do.

---

I am throwing an error now in case of price is not available from Oracle.